.github/workflows/release.yml

```diff
@@ -1,16 +1,37 @@
-name: release
+name: 'New Release'
+run-name: 'Release ${{ inputs.version_number }}'
+
 on:
-  push:
-    tags:
-      - 'v*'
-jobs: 
-  goreleaser:
+  workflow_dispatch:
+    inputs:
+      version_number:
+        description: 'Version number (e.g., v1.0.0, v1.0.0-pre, v1.0.0-pre1)'
+        required: true
+      skip_tests:
+        description: 'Skip QA acceptance tests, define value to `true` to explicitly skip'
+
+jobs:
+    # QA acceptance tests are skipped for pre releases or when explicit input parameter is used
+  run-qa-acceptance-tests:
+    if: inputs.skip_tests != 'true' && !contains(inputs.version_number, 'pre')
+    secrets: inherit
+    uses: ./.github/workflows/acceptance-tests.yml
+    with:
+      atlas_cloud_env: "qa"
+
+  release:
     runs-on: ubuntu-latest
+    needs: [ run-qa-acceptance-tests ]
+    if: ${{ always() && (needs.run-qa-acceptance-tests.result == 'skipped' || needs.run-qa-acceptance-tests.result == 'success') }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Unshallow
         run: git fetch --prune --unshallow
+      - name: Create release tag
+        uses: rickstaa/action-create-tag@v1.7.2 # will fail if existing tag is present
+        with:
+          tag: ${{ inputs.version_number }}
```

Comment:
as of now tag is not being signed, which is the behaviour we currently have when [creating tags for our releases](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/RELEASING.md#pre-release-the-provider). Will reach out to security team to verify if we can use the gpg signing feature of this action.

Reply:
yes verified that annotated tags were created in the forked repo used for testing, can also be seen in the code https://github.com/rickstaa/action-create-tag/blob/main/entrypoint.sh#L79